File: example/java-example/src/main/java/com/spring/example/model/BaseResponseMapper.java
package com.spring.example.model;


import com.spring.example.openapi.model.BaseResponseDTO;
import org.mapstruct.Mapper;

@Mapper(componentModel = "java")
public interface BaseResponseMapper {

    BaseResponse parse(BaseResponseDTO input);

    BaseResponseDTO parse(BaseResponse input);
}

File: example/kotlin-example/src/main/kotlin/com/spring/example/model/BaseResponseMapper.kt
package com.spring.example.model

import com.spring.example.openapi.model.BaseResponseDTO
import org.mapstruct.Mapper

@Mapper(componentModel = "java")
interface BaseResponseMapper {
    fun parse(input: BaseResponseDTO): BaseResponse
    fun parse(input: BaseResponse): BaseResponseDTO
}
